packages/mobile/src/components/core/PullToRefresh.tsx

```diff
@@ -218,16 +245,23 @@ export const PullToRefresh = ({
           onRefresh?.()
           animationRef.current?.play()
         }
+        if (value > 0 && didHitTop) {
+          setShouldShowRefresh(false)
+        }
+
+        if (value < -20 && didHitTop) {
+          setShouldShowRefresh(true)
+        }
       }
     )
     return () => {
       if (listenerRef.current) {
         scrollAnim?.removeListener(listenerRef.current)
       }
     }
-  }, [scrollAnim, onRefresh, isRefreshDisabled])
+  }, [scrollAnim, onRefresh, isRefreshDisabled, didHitTop])
 
-  return scrollAnim ? (
+  return scrollAnim && shouldShowRefresh ? (
```

Comment:
rather than a boolean to gate it, does it make sense to tweak opacity with animated so it happens outside of the JS loop? (and then to hide it)

Reply:
oh yeah that's smart, yeah can try that out

Reply:
updated this, great call